### DISCUSSION TITLE
Deleted model still appears in blade (foreach)

### DISCUSSION THREAD

**Discussion by rdmngithub** [livewire/livewire#4931] [1. Help] [answered]
Hello,
I have a problem with a collection in a foreach:
```
@foreach (Auth::user()->paymentMethods as $paymentMethod)
 <li>{{ $paymentMethod->name }}</li>
@endforeach
```
In a method when I do:
`Auth::user()->paymentMethods()->where('id', 1)->delete();`
the concerned paymentMethod is deleted in database and no longer displayed in blade

But when I put a condition with if:
```
if ($condition) {
Auth::user()->paymentMethods()->where('id', 1)->delete();
}
```
the concerned paymentMethod is deleted in database but **still appears in blade**
Does anyone know the cause of the problem? Thanks in advance

**Comment by NBA707** [comment]
Ugly but?

@if (! $condition) {
   <li>{{ $paymentMethod->name }}</li>
@endif

**Comment by joshhanley** [comment, marked as answer]
@rdmngithub you are using `Auth::user()->paymentMethods` which if payment methods are already loaded into memory, then it may show the deleted one.

Try changing to `Auth::user()->paymentMethods()->get()` and that should get the latest directly from the database.

Hope this helps!
